StoreRecoverer transformed in RecoveryRequiredChecker

davidegrohmann opened the pull request:
Since the recovery functionality present in StoreRecover is not needed any longer, it has been dropped.  In such a way the component is responsible only for checking if recovery is required on a given database.

jotomo commented on community/kernel/src/main/java/org/neo4j/kernel/impl/recovery/StoreRecoveryChecker.java:
```diff
@@ -36,24 +33,20 @@
+ * An external tool that can determine if a given store will need recovery.
  */
-public class StoreRecoverer
+public class StoreRecoveryChecker
```
Hm, I wonder if _StoreStateChecker_ would be a better name. _StoreRecoveryChecker_ might also be the name of a class that checks if a recovery has been successful.

MishaDemianenko commented on community/kernel/src/main/java/org/neo4j/kernel/impl/recovery/StoreRecoveryChecker.java:
```diff
@@ -36,24 +33,20 @@
     }
 
-    public boolean recoveryNeededAt( File dataDir ) throws IOException
+    public boolean recoveryNeededAt( File dataDir, PageCache pageCache ) throws IOException
```
Can we rename method to `isRecoveryNeededAt` or smt. simular?

thobe commented:
The title of the commit/PR is a bit poor, borderline misleading, but the changes look sensible. I'd like it if the commit message contained a bit more details about the motivation for this change as well.

chrisvest commented:
@davidegrohmann Needs rebase.

davidegrohmann committed:
StoreRecoverer transformed in RecoveryRequiredChecker

Since the recovery functionality present in StoreRecover is not needed
any longer, it has been dropped.  In such a way the component is
responsible only for checking if recovery is required on a given
database.
